test-data/unit/check-classes.test

```diff
@@ -6512,3 +6512,27 @@ def access_after_declaration(self) -> None:
 
     reveal_type(x)  # N: Revealed type is 'builtins.int'
     x = x + 1
+
+[case testIsSubClassNarrowDownTypesOfTypeVariables]
+from typing import TypeVar, Generic
+
+TypeT = TypeVar("TypeT", bound=type)
+
+class Base:
+    field: int = 42
+
+class C1:
+    def method(self, other: type) -> int:
+        if issubclass(other, Base):
+            reveal_type(other)  # N: Revealed type is 'Type[__main__.Base]'
+            return other.field
+        return 0
+
+class C2(Generic[TypeT]):
+    def method(self, other: TypeT) -> int:
+        if issubclass(other, Base):
+            reveal_type(other)  # N: Revealed type is 'Type[__main__.Base]'
+            return other.field
+        return 0
+
+[builtins fixtures/isinstancelist.pyi]
```

Comment:
OK, now the two classes don't look the same, but you didn't do what I asked, I wanted type variables with different upper bounds, like `Type[SomeClass]`, not just `type`.